### PR TITLE
che #14947: Releasing 7.3.1 version of the che-machine-exec. Adding rhel.Dockerfile

### PR DESCRIPTION
**Pull request by ibuziuk**
Che issue - https://github.com/eclipse/che/issues/14947
Adding rhel.Dockerfile from:
- https://github.com/eclipse/che-machine-exec/pull/49
- https://github.com/eclipse/che-machine-exec/commit/19c1fed64fcecf56b10a8500f02869b889231974

### CODE REVIEW

File: rhel.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+#
+
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/rhel8/go-toolset
+FROM rhel8/go-toolset:1.11.13-18 as builder
```
Comment (benoitf) [inline rhel.Dockerfile]: this image is not public it can't be used in Eclipse Che

Comment (ibuziuk) [reply rhel.Dockerfile]: @nickboldt I have no idea why this image was added in general without proper ci setup? Could you clarify?

Comment (nickboldt) [reply rhel.Dockerfile]: it's not used in Che. It's a secondary image that makes it easier for upstream to remember that there's a downstream, so if they update Dockerfile they MIGHT remember to apply similar change to rhel.Dockerfile. But it's not strictly an artifact that's USED in the Che build process

Comment (ibuziuk) [reply rhel.Dockerfile]: yeah, but for devfile & plugin registry we do build rhel image as part of CI. Does it make sense to do the same for `che-machine-exec`. In the current state presense of this  RHEL image is misleading IMO

Comment (nickboldt) [reply rhel.Dockerfile]: well then there are two options:

a) keep it up here i che-machine-exec and DO build it as part of CI

b) remove it and hope that not having a rhel version in upstream causes any downstream impacts when I have to absorb changes from up to down and can't find an SME online (due to TZ conflicts).

I'd prefer (a). In fact I'd prefer if building on UBI8 was the default, not an option, but hey. :)

**Pull request merged**